doc/specs/#1571 - New Tab Menu Customization/#1571 - New Tab Menu Customization.md

```diff
@@ -202,6 +213,13 @@ The design chosen in this spec more cleanly separates the responsibilities of
 the list of profiles and the contents of the new tab menu. This way, each object
 can be defined independent of the structure of the other.
 
+Regarding implementation of `matchProfile` entries: In order to build the menu,
+we'll evaluate the entries in the following order:
+
+* all explicit `profile` entries
+* then all `matchProfile` entries, using profiles not already specified

```

Comment:
We should also specify the order of evaluation of these entries, otherwise the behaviour of `matchProfile: name: .*` in the main menu and `matchProfile: source: Wsl` in a folder would not be defined. (I think the only two options are depth-first or breadth-first of which I think depth-first is the optimal implementation logic)

Or do we want to add a flag toggling the "remaining" feature? So we can have this entry act both as "searching all that match unconditionally", or "search all that match that have not yet been specified elsewhere".

Reply:
I'm all for giving people more settings than they could all possibly use. At this point I think I'd love to chase down a sensible default first, in the case we never get around to the setting to toggle it. 

Maybe it should be a property after all? Maybe default to false, cause true seems harder to implement. When it's true, we'd do a DFS as we evaluate the rest of the `matchProfile` and `remainingProfiles` entries. 

(idle thought: we may want to consider the fact that `matchProfile(name:".*", remaningOnly:true)` === `remainingProfiles`)

Reply:
I'm really in doubt whether `remainingOnly: true` is a sensible default at all. Intuitively, `matchProfile` should always return all matches and should not differ depending on where I put the entry in my menu.

And then, since indeed remainingProfiles is not "unique", we might just as well implement this flag immediately.

Reply:
Okay, I think this should be clarified in the latest version